src/utils/update/beneficial_owners_managing_officers_data_fetch.ts

```diff
@@ -58,13 +60,33 @@ export const retrieveBeneficialOwners = async (req: Request, appData: Applicatio
 };
 
 export const retrieveManagingOfficers = async (req: Request, appData: ApplicationData) => {
+  const transactionId = appData.transaction_id;
+  const overseasEntityId = appData.overseas_entity_id;
   const companyOfficers = await getCompanyOfficers(req, appData[EntityNumberKey] as string);
+  let moPrivateData: ManagingOfficersPrivateData = {
+    moPrivateData: []
+  };
+
+  try {
+    if (transactionId && overseasEntityId) {
+      moPrivateData = await getManagingOfficerPrivateData(req, transactionId, overseasEntityId) as ManagingOfficersPrivateData;
```

Comment:
I don't think you get any moPrivateData when it's a 404? getManagingOfficerPrivateData returns resource.response but I would expect that to be undefined or empty. Maybe clearer to change getManagingOfficerPricateData (or use it defensively).

Reply:
line 73 assumes you got moPrivateData.moPrivateData

Reply:
I'll add this: if (!moPrivateData || moPrivateData.moPrivateData.length === 0)

Reply:
drop the "as ManagingOfficerPrivateData"